modules/phase_field/include/userobjects/RandomEulerAngleProvider.h

```diff
@@ -0,0 +1,43 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef RANDOMEULERANGLEPROVIDER_H
+#define RANDOMEULERANGLEPROVIDER_H
+
+#include "EulerAngleProvider.h"
+#include "GrainTrackerInterface.h"
+#include "EulerAngles.h"
+#include "MooseRandom.h"
```

Comment:
Prefer forward declaration and then include this in the .C file. Helps with speed in some cases.